### PR TITLE
[Move Spring beans]Lps 201149

### PR DESCRIPTION
**Pull request by slnn**
https://liferay.atlassian.net/browse/LPS-201149

### CODE REVIEW

**Review by dantewang — APPROVED** [review]

**Comment by dantewang** [comment]
@slnn Conflicts, please update.

**Comment by slnn** [comment]
> @slnn Conflicts, please update.

Done

**Comment by dantewang** [comment]
ci:test:relevant

**Comment by dantewang** [comment]
ci:test:sf

**Comment by Kvale1733** [comment]
ci:test:relevant

**Comment by zhangyan0701** [comment]
Semantic versioning issue:
```
     [exec] > Task :baseline
     [exec]   PACKAGE_NAME                                       DELTA      CUR_VER    BASE_VER   REC_VER    WARNINGS  
     [exec] = ================================================== ========== ========== ========== ========== ==========
     [exec] * com.liferay.portal.kernel.portlet                  MAJOR      28.0.0     28.0.0     29.0.0     VERSION INCREASE REQUIRED
     [exec] 	>   class      com.liferay.portal.kernel.portlet.PortletQNameUtil
     [exec] 		+   field      PRIVATE_RENDER_PARAMETER_NAMESPACE
     [exec] 			+   access     final
     [exec] 			+   access     static
     [exec] 			+   constant   priv_r_p_
     [exec] 			+   return     java.lang.String
     [exec] 		+   field      PUBLIC_RENDER_PARAMETER_NAMESPACE
     [exec] 			+   access     final
     [exec] 			+   access     static
     [exec] 			+   constant   p_r_p_
     [exec] 			+   return     java.lang.String
     [exec] 		+   field      REMOVE_PUBLIC_RENDER_PARAMETER_NAMESPACE
     [exec] 			+   access     final
     [exec] 			+   access     static
     [exec] 			+   constant   r_p_r_p_
     [exec] 			+   return     java.lang.String
     [exec] 		-   method     getPortletQName()
     [exec] 			-   access     static
     [exec] 			-   return     com.liferay.portal.kernel.portlet.PortletQName
     [exec] 		-   method     getPublicRenderParameterIdentifier(java.lang.String)
     [exec] 			-   access     static
     [exec] 			-   return     java.lang.String
     [exec] 		-   method     setPortletQName(com.liferay.portal.kernel.portlet.PortletQName)
     [exec] 			-   return     void
     [exec] 	-   interface  com.liferay.portal.kernel.portlet.PortletQName
     [exec] 		-   access     abstract
     [exec] 		-   field      PRIVATE_RENDER_PARAMETER_NAMESPACE
     [exec] 			-   access     final
     [exec] 			-   access     static
     [exec] 			-   constant   priv_r_p_
     [exec] 			-   return     java.lang.String
     [exec] 		-   field      PUBLIC_RENDER_PARAMETER_NAMESPACE
     [exec] 			-   access     final
     [exec] 			-   access     static
     [exec] 			-   constant   p_r_p_
     [exec] 			-   return     java.lang.String
     [exec] 		-   field      REMOVE_PUBLIC_RENDER_PARAMETER_NAMESPACE
     [exec] 			-   access     final
     [exec] 			-   access     static
     [exec] 			-   constant   r_p_r_p_
     [exec] 			-   return     java.lang.String
     [exec] 		-   method     getKey(com.liferay.portal.kernel.xml.QName)
     [exec] 			-   access     abstract
     [exec] 			-   return     java.lang.String
     [exec] 		-   method     getKey(java.lang.String,java.lang.String)
     [exec] 			-   access     abstract
     [exec] 			-   return     java.lang.String
     [exec] 		-   method     getPublicRenderParameterIdentifier(java.lang.String)
     [exec] 			-   access     abstract
     [exec] 			-   return     java.lang.String
     [exec] 		-   method     getPublicRenderParameterName(com.liferay.portal.kernel.xml.QName)
     [exec] 			-   access     abstract
     [exec] 			-   return     java.lang.String
     [exec] 		-   method     getQName(com.liferay.portal.kernel.xml.Element,com.liferay.portal.kernel.xml.Element,java.lang.String)
     [exec] 			-   access     abstract
     [exec] 			-   return     com.liferay.portal.kernel.xml.QName
     [exec] 		-   method     getQName(java.lang.String)
     [exec] 			-   access     abstract
     [exec] 			-   return     com.liferay.portal.kernel.xml.QName
     [exec] 		-   method     getRemovePublicRenderParameterName(com.liferay.portal.kernel.xml.QName)
     [exec] 			-   access     abstract
     [exec] 			-   return     java.lang.String
     [exec] 		-   method     setPublicRenderParameterIdentifier(java.lang.String,java.lang.String)
     [exec] 			-   access     abstract
     [exec] 			-   return     void
     [exec] [Baseline Report] Mode: diff (persisted)
     [exec] Semantic versioning is incorrect while checking /opt/dev/projects/github/liferay-portal/portal-kernel/portal-kernel.jar against /opt/dev/projects/github/liferay-portal/.gradle/caches/modules-2/files-2.1/com.liferay.portal/com.liferay.portal.kernel/129.1.0/54686f393e5624f23b69431a979085bcbd7e2cb6/com.liferay.portal.kernel-129.1.0.jar
```
Full console log:
https://testray.liferay.com/reports/production/logs/2023-11/test-1-35/test-portal-acceptance-pullrequest(master)/4957/semantic-versioning-jdk8/0/0/jenkins-console.txt.gz?authuser=0

**Comment by slnn** [comment]
> Semantic versioning issue:
> 
> ```
>      [exec] > Task :baseline
>      [exec]   PACKAGE_NAME                                       DELTA      CUR_VER    BASE_VER   REC_VER    WARNINGS  
>      [exec] = ================================================== ========== ========== ========== ========== ==========
>      [exec] * com.liferay.portal.kernel.portlet                  MAJOR      28.0.0     28.0.0     29.0.0     VERSION INCREASE REQUIRED
>      [exec] 	>   class      com.liferay.portal.kernel.portlet.PortletQNameUtil
>      [exec] 		+   field      PRIVATE_RENDER_PARAMETER_NAMESPACE
>      [exec] 			+   access     final
>      [exec] 			+   access     static
>      [exec] 			+   constant   priv_r_p_
>      [exec] 			+   return     java.lang.String
>      [exec] 		+   field      PUBLIC_RENDER_PARAMETER_NAMESPACE
>      [exec] 			+   access     final
>      [exec] 			+   access     static
>      [exec] 			+   constant   p_r_p_
>      [exec] 			+   return     java.lang.String
>      [exec] 		+   field      REMOVE_PUBLIC_RENDER_PARAMETER_NAMESPACE
>      [exec] 			+   access     final
>      [exec] 			+   access     static
>      [exec] 			+   constant   r_p_r_p_
>      [exec] 			+   return     java.lang.String
>      [exec] 		-   method     getPortletQName()
>      [exec] 			-   access     static
>      [exec] 			-   return     com.liferay.portal.kernel.portlet.PortletQName
>      [exec] 		-   method     getPublicRenderParameterIdentifier(java.lang.String)
>      [exec] 			-   access     static
>      [exec] 			-   return     java.lang.String
>      [exec] 		-   method     setPortletQName(com.liferay.portal.kernel.portlet.PortletQName)
>      [exec] 			-   return     void
>      [exec] 	-   interface  com.liferay.portal.kernel.portlet.PortletQName
>      [exec] 		-   access     abstract
>      [exec] 		-   field      PRIVATE_RENDER_PARAMETER_NAMESPACE
>      [exec] 			-   access     final
>      [exec] 			-   access     static
>      [exec] 			-   constant   priv_r_p_
>      [exec] 			-   return     java.lang.String
>      [exec] 		-   field      PUBLIC_RENDER_PARAMETER_NAMESPACE
>      [exec] 			-   access     final
>      [exec] 			-   access     static
>      [exec] 			-   constant   p_r_p_
>      [exec] 			-   return     java.lang.String
>      [exec] 		-   field      REMOVE_PUBLIC_RENDER_PARAMETER_NAMESPACE
>      [exec] 			-   access     final
>      [exec] 			-   access     static
>      [exec] 			-   constant   r_p_r_p_
>      [exec] 			-   return     java.lang.String
>      [exec] 		-   method     getKey(com.liferay.portal.kernel.xml.QName)
>      [exec] 			-   access     abstract
>      [exec] 			-   return     java.lang.String
>      [exec] 		-   method     getKey(java.lang.String,java.lang.String)
>      [exec] 			-   access     abstract
>      [exec] 			-   return     java.lang.String
>      [exec] 		-   method     getPublicRenderParameterIdentifier(java.lang.String)
>      [exec] 			-   access     abstract
>      [exec] 			-   return     java.lang.String
>      [exec] 		-   method     getPublicRenderParameterName(com.liferay.portal.kernel.xml.QName)
>      [exec] 			-   access     abstract
>      [exec] 			-   return     java.lang.String
>      [exec] 		-   method     getQName(com.liferay.portal.kernel.xml.Element,com.liferay.portal.kernel.xml.Element,java.lang.String)
>      [exec] 			-   access     abstract
>      [exec] 			-   return     com.liferay.portal.kernel.xml.QName
>      [exec] 		-   method     getQName(java.lang.String)
>      [exec] 			-   access     abstract
>      [exec] 			-   return     com.liferay.portal.kernel.xml.QName
>      [exec] 		-   method     getRemovePublicRenderParameterName(com.liferay.portal.kernel.xml.QName)
>      [exec] 			-   access     abstract
>      [exec] 			-   return     java.lang.String
>      [exec] 		-   method     setPublicRenderParameterIdentifier(java.lang.String,java.lang.String)
>      [exec] 			-   access     abstract
>      [exec] 			-   return     void
>      [exec] [Baseline Report] Mode: diff (persisted)
>      [exec] Semantic versioning is incorrect while checking /opt/dev/projects/github/liferay-portal/portal-kernel/portal-kernel.jar against /opt/dev/projects/github/liferay-portal/.gradle/caches/modules-2/files-2.1/com.liferay.portal/com.liferay.portal.kernel/129.1.0/54686f393e5624f23b69431a979085bcbd7e2cb6/com.liferay.portal.kernel-129.1.0.jar
> ```
> 
> Full console log: https://testray.liferay.com/reports/production/logs/2023-11/test-1-35/test-portal-acceptance-pullrequest(master)/4957/semantic-versioning-jdk8/0/0/jenkins-console.txt.gz?authuser=0

Thank @zhangyan0701  for catching this! Fix and update this pull.

**Comment by slnn** [comment]
ci:test:relevant

**Comment by slnn** [comment]
ci:test:sf

**Comment by Kvale1733** [comment]
@slnn No regressions found.

File: portal-impl/src/com/liferay/portlet/internal/PortletQNameImpl.java [hunk 1]
```diff
@@ -37,21 +37,6 @@ public String getKey(String uri, String localPart) {
 		return StringBundler.concat(uri, _KEY_SEPARATOR, localPart);
 	}
 
-	@Override
-	public String getPublicRenderParameterIdentifier(
```
Comment (dantewang) [inline portal-impl/src/com/liferay/portlet/internal/PortletQNameImpl.java:40]: @slnn Can you keep this method? If removing, it needs to be removed together with `setPublicRenderParameterIdentifier` which still has usage. Please double check whether this identifier still makes any sense.

**Review by dantewang — CHANGES_REQUESTED** [review]
@slnn Please resend, and
1) Rebase sem ver
2) Update breaking change messages to follow the new format (no empty lines between texts, etc.)

File: portal-impl/src/com/liferay/portlet/internal/PortletQNameImpl.java (hunk 1 above)
Comment (slnn) [reply portal-impl/src/com/liferay/portlet/internal/PortletQNameImpl.java:40]: Hi @dantewang 

I have double-checked, and we can remove only one usage of `setPublicRenderParameterIdentifier()` and then remove the method as no one will use the identifier after removing the `getPublicRenderParameterIdentifier()`

**Comment by slnn** [comment]
Please see https://github.com/liferay-core-infra/liferay-portal/pull/6366

**Pull request closed without merging**